### ISSUE TITLE
Warning and error given by VS Code while obsoleting multiple tables that are having relation.

### ISSUE THREAD

**Issue by CML1987** [microsoft/AL#5936]
I am obsoleting multiple tables that are having relation, the VS Code is throwing warning or error although the table that is having the warning or error is having the ObsoleteState set to Pending or Removed. The warning or error will stay there forever unless I remove all the relevant variable and TableRelations, this has destroyed the purpose of obsoleting the object because we can't keep the code and we need to perform the code clean up once we set it to Removed state.

**To Reproduce**
Steps and to reproduce the behavior:
1. Create 2 Tables, Header and Line.
2. Set the TableRelation between the 2 tables.
3. Set the ObsoleteState to Pending / Removed for both tables.

``` 
table 50100 Header
{
    ObsoleteState = Pending;

    fields
    {
        field(1; "No."; Code[20])
        {
            Caption = 'No.';
        }
    }

    keys
    {
        key(PK; "No.")
        {
            Clustered = true;
        }
    }

    procedure GetNo(var No: Code[20])
    begin
        No := "No.";
    end;
}

```

```
table 50101 Line
{
    ObsoleteState = Pending;

    fields
    {
        field(1; "Header No."; Code[20])
        {
            Caption = 'Header No.';
            TableRelation = Header."No.";
        }
        field(2; "No."; Code[20])
        {
            Caption = 'No.';
        }
    }

    keys
    {
        key(PK; "Header No.", "No.")
        {
            Clustered = true;
        }
    }

    local procedure GetHeaderNo(): Code[20]
    var
        HeaderNo: Code[20];
        Header: Record Header;
    begin
        Header.GetNo(HeaderNo);
        exit(HeaderNo);
    end;
}
```

**Expected behavior**
The warning should not be thrown if we have obsoleted object that having the variable / functions from another obsoleted object.

**Screenshots**
![image](https://user-images.githubusercontent.com/66260981/83537804-a3e19900-a527-11ea-9864-b341490ff837.png)

![image](https://user-images.githubusercontent.com/66260981/83538684-b8726100-a528-11ea-9af0-7bb0e6ced02a.png)

**5. Versions:**

 - AL Language: 5.0.280447

 - Business Central:

**Comment by kalberes** [comment]
You mean that the warning does not make any sense:
Table 'Header' is removed.  This warning will become an error in a future release.

since there is already an error message:
Table 'Header' is removed. 

Agree.

**Comment by CML1987** [comment]
Nope, the warning is caused by the TableRelation in field "Header No." of "Line" and the error is due to the table variable defined in the table "Line".

The warning and error in table "Line" does not make any sense since I have set the ObsoleteState of Table "Line" to "Removed". 

The obsolete related warning and error should not be thrown if,

"Header" is set to ObsoleteState = Pending
"Line" is set to ObsoleteState = "Pending" or "Removed".

or

Both tables are set to ObsoleteState = Removed.

**Comment by kalberes** [comment]
For now TableRelations do not participate in  Obsolete handling and probably will not in the nearby future

**Comment by CML1987** [comment]
Sorry, maybe I didn't describe it very nicely

Yes, I'm binding something that is still pending to remove. But the caller object, "Line" in this case has also been set to pending to remove. So it means that we want to remove both of these table in the future...and maybe my screen shot and the sample code has confused you. Please look at the new screenshots...

![image](https://user-images.githubusercontent.com/66260981/83830697-55472100-a718-11ea-81b6-cee74eb5ddee.png)

![image](https://user-images.githubusercontent.com/66260981/83830709-61cb7980-a718-11ea-81c0-231528ef1249.png)

In the new screenshots, you will see that both tables have been set to Removed state but still the error and warning are showing. In order to make the PROBLEM tab looks clean, I need to remove all the tablerelation and variables that are having error/warning. Is this the correct way of doing so? Because in my understanding, the ObsoleteState in the object is a method for us to obsolete the object and restrict the user from using it without removing of a single line of code.

**Comment by salgiza** [comment]
Just for reference, this is another instance of the problem that I described a few months ago here: https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=613940878835712 related to the new sales price functionality of BC. Every object that we had extended is now marked obsolete, and we get lots of warnings even though our own code is marked as obsolete.

As @CML1987 mentions, it makes no sense that something obsolete raises a warning because it references another thing that is also obsolete.
